WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadViewModel.kt

```diff
@@ -1,15 +1,130 @@
+const val KEY_BACKUP_DOWNLOAD_COMPLETED = "key_backup_download_completed"
+const val KEY_BACKUP_DOWNLOAD_STATE = "key_backup_download_state"
+
+@Parcelize
+@SuppressLint("ParcelCreator")
+data class BackupDownloadState(
+    val siteId: Long? = null,
+    val rewindId: Long? = null
+) : WizardState, Parcelable
+
+typealias NavigationTarget = WizardNavigationTarget<BackupDownloadStep, BackupDownloadState>
+
+class BackupDownloadViewModel @Inject constructor(
+    private val wizardManager: WizardManager<BackupDownloadStep>
+) : ViewModel() {
+    private var isStarted = false
+    private var backupDownloadCompleted = false
+
+    private lateinit var backupDownloadState: BackupDownloadState
+
+    val navigationTargetObservable: SingleEventObservable<NavigationTarget> by lazy {
+        SingleEventObservable(
+                Transformations.map(wizardManager.navigatorLiveData) {
+                    clearOldBackupDownloadState(it)
+                    WizardNavigationTarget(it, backupDownloadState)
+                }
+        )
+    }
 
+    private val _wizardFinishedObservable = SingleLiveEvent<BackupDownloadWizardState>()
+    val wizardFinishedObservable: LiveData<BackupDownloadWizardState> = _wizardFinishedObservable
+
+    private val _screenTitleObservable = SingleLiveEvent<@StringRes Int>()
+    val screenTitleObservable: LiveData<Int> = _screenTitleObservable
+
+    private val _exitFlowObservable = SingleLiveEvent<Unit>()
+    val exitFlowObservable: LiveData<Unit> = _exitFlowObservable
+
+    private val _onBackPressedObservable = SingleLiveEvent<Unit>()
+    val onBackPressedObservable: LiveData<Unit> = _onBackPressedObservable
+
+    fun start(savedInstanceState: Bundle?) {
+        if (isStarted) return
+        if (savedInstanceState == null) {
+            backupDownloadState = BackupDownloadState()
+        } else {
+            backupDownloadCompleted = savedInstanceState.getBoolean(
+                    KEY_BACKUP_DOWNLOAD_COMPLETED,
+                    false
+            )
+            backupDownloadState = requireNotNull(
+                    savedInstanceState.getParcelable(
+                            KEY_BACKUP_DOWNLOAD_STATE
+                    )
+            )
+            val currentStepIndex = savedInstanceState.getInt(KEY_BACKUP_DOWNLOAD_CURRENT_STEP)
+            wizardManager.setCurrentStepIndex(currentStepIndex)
+        }
         isStarted = true
+        if (savedInstanceState == null) {
+            // Show the next step only if it's a fresh activity so we can handle the navigation
+            wizardManager.showNextStep()
+        }
+    }
+
+    fun writeToBundle(outState: Bundle) {
+        outState.putBoolean(KEY_BACKUP_DOWNLOAD_COMPLETED, backupDownloadCompleted)
+        outState.putInt(KEY_BACKUP_DOWNLOAD_CURRENT_STEP, wizardManager.currentStep)
+        outState.putParcelable(KEY_BACKUP_DOWNLOAD_STATE, backupDownloadState)
+    }
+
+    fun onBackPressed() {
```

Comment:
1) Minor (🔍 ): Consider making this a one-liner and move the `TODO` comment like below.
```
    // todo: annmarie - what should happen on backPress - always exit, please revisit
    fun onBackPressed() = exitFlow()
```
2) Minor (🔍 ): Since `exitFlow()` is used only here, consider moving this function closer to usage. Also, why not using `_exitFlowObservable.call()` directly?

Reply:
> Since exitFlow() is used only here, consider moving this function closer to usage. Also, why not using _exitFlowObservable.call() directly?

Lots of tweaking coming to this class ... it may not be the only place that uses exitFlow(). I'll note this for later. thanks.